File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/psychotypeDescription/PsychotypeDescriptionAdapter.java
package com.laushkina.anastasia.psychosophy.view.psychotypeDescription;

import android.app.Fragment;
import android.app.FragmentManager;
import android.os.Bundle;
import android.support.v13.app.FragmentPagerAdapter;

import com.laushkina.anastasia.psychosophy.domain.Psychotype;
import com.laushkina.anastasia.psychosophy.view.psychotypes.PsychotypesFragment;

public class PsychotypeDescriptionAdapter extends FragmentPagerAdapter {

    public static final int PREVIEW_DESCRIPTION = 0;
    public static final int FULL_TEXT_DESCRIPTION = 1;
    private static final int PAGE_AMOUNT = 2;

    private Psychotype psychotype;

    PsychotypeDescriptionAdapter(FragmentManager fragmentManager, Psychotype psychotype){
        super(fragmentManager);
        this.psychotype = psychotype;
    }

    public void setPsychotype(Psychotype psychotype){
        this.psychotype = psychotype;
        notifyDataSetChanged();
    }

    @Override
    public Fragment getItem(int position) {
        Fragment fragment;
        switch (position){
            case PREVIEW_DESCRIPTION: {
                fragment = new PsychotypePreviewDescriptionFragment();
                break;
            }
            case FULL_TEXT_DESCRIPTION: {
                fragment = new PsychotypeFullTextDescriptionFragment();
                break;
            }
            default:
                throw new RuntimeException("Cannot detect psychotype description page");
        }

        Bundle typeExtra = new Bundle(1);
        typeExtra.putSerializable(PsychotypesFragment.psychotypeExtra, psychotype);
        fragment.setArguments(typeExtra);

        return fragment;
    }

    @Override
    public int getCount() {
        return PAGE_AMOUNT;
    }
}

File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/psychotypeDescription/PsychotypeDescriptionAdapter.kt
package com.laushkina.anastasia.psychosophy.view.psychotypeDescription

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.laushkina.anastasia.psychosophy.domain.Psychotype
import com.laushkina.anastasia.psychosophy.view.psychotypes.PsychotypesFragment

class PsychotypeDescriptionAdapter(fm: FragmentManager,
                                   private var psychotype: Psychotype?): FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    fun setPsychotype(psychotype: Psychotype) {
        this.psychotype = psychotype
        notifyDataSetChanged()
    }

    override fun getItem(position: Int): Fragment {
        val fragment: Fragment = when (position) {
            PREVIEW_DESCRIPTION -> {
                PsychotypeOverviewDescriptionFragment()
            }
            FULL_TEXT_DESCRIPTION -> {
                PsychotypeFullTextDescriptionFragment()
            }
            else -> throw RuntimeException("Cannot detect psychotype description page")
        }

        val typeExtra = Bundle(1)
        typeExtra.putSerializable(PsychotypesFragment.psychotypeExtra, psychotype)
        fragment.arguments = typeExtra

        return fragment
    }

    override fun getCount(): Int {
        return PAGE_AMOUNT
    }

    companion object {
        const val PREVIEW_DESCRIPTION = 0
        const val FULL_TEXT_DESCRIPTION = 1
        private const val PAGE_AMOUNT = 2
    }

}